stdlib/test/builtin/test_float_literal.mojo

```diff
@@ -192,6 +192,11 @@ def test_abs():
     assert_equal(FloatLiteral.__abs__(neg_inf), inf)
 
 
+def test_float_conversion():
+    assert_equal(float(4.5), 4.5)
+    assert_equal(float(0.0), 0.0)
```

Comment:
See me other comment; this might not actually test `FloatLiteral.__float__` due to implicit conversion and materialization.